.ci/Jenkinsfile

```diff
@@ -39,10 +39,10 @@ pipeline {
             withMageEnv(){
               sh(label: 'lint', script: '''
                 go mod tidy && git diff --exit-code
-                gofmt -l . | read && echo "Code differs from gofmt's style. Run 'gofmt -w .'" 1>&2 && exit 1 || true
+								curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.44.0
```

Comment:
Given that we are adding more and more linting to this step, it would be better to extract it into a separate mage step. Maybe it can be added to `mage check` and we could remove this step from the CI.

Reply:
+ 1 to a separate step.

Reply:
@ph @kvch 
but does it really make any difference to have the linter as a separate step in Jenkins (as it is now) or as a separate step in the Magefile?

Mind you, I have not added more linting to the existing step, I replaced the linter we already had with a new one (because it includes the same checks, so we don't need the old one anymore).

Reply:
Good point, I'll move it to the magefile.

Reply:
Addressed in af131c82958bf60e3f1c1a5afd480b5807090b5b